tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java

```diff
@@ -257,6 +257,10 @@ private long readLongLittleEndian(CodecDataInput cdi) {
         + ((long) (readBuffer[7] & 255) << 56));
   }
 
+  public boolean isSameCatalog(DataType other) {
```

Comment:
how about rename to shouldNarrowDataTypeTo?

Reply:
By `isSameCatalog`, I mean two different data type can be compared directly if them are in the same catalog, so I think `isSameCatalog` is better than `shouldNarrowDataTypeTo `